p2p/discovery.py

```diff
@@ -765,6 +791,45 @@ def process(received_echo: Hash32, response: List[kademlia.Node]) -> None:
 
         return tuple(n for n in nodes if n != self.this_node)
 
+    async def get_ticket(self, node: kademlia.Node, topics: List[bytes]) -> 'Ticket':
```

Comment:
See my comment below about return value checking but.... the signature for this function says it returns a `Ticket` when it should really be `Optional[Ticket]` right?

Reply:
That's right